src/ert/shared/ensemble_evaluator/evaluator.py

```diff
@@ -209,32 +209,45 @@ async def handle_dispatch(self, websocket, path):
         # pylint: disable=not-async-context-manager
         # (false positive)
         async with self.count_dispatcher():
-            async for msg in websocket:
-                try:
-                    event = from_json(msg, data_unmarshaller=evaluator_unmarshaller)
-                except cloudevents.exceptions.DataUnmarshallerError:
-                    event = from_json(msg, data_unmarshaller=pickle.loads)
-                if self._get_ens_id(event["source"]) != self.ensemble.id_:
-                    logger.info(
-                        f"Got event from evaluator {self._get_ens_id(event['source'])} "
-                        f"with source {event['source']}, "
-                        f"ignoring since I am {self.ensemble.id_}"
-                    )
-                    continue
-                try:
-                    await self._dispatcher.handle_event(event)
-                except BaseException as ex:
-                    logger.warning(
-                        f"cannot handle event - closing connection to dispatcher: {ex}"
-                    )
-                    await websocket.close(code=1011, reason=f"failed handling {event}")
-                    return
-
-                if event["type"] in [
-                    identifiers.EVTYPE_ENSEMBLE_STOPPED,
-                    identifiers.EVTYPE_ENSEMBLE_FAILED,
-                ]:
-                    return
+            try:
+                async for msg in websocket:
+                    try:
+                        event = from_json(msg, data_unmarshaller=evaluator_unmarshaller)
+                    except cloudevents.exceptions.DataUnmarshallerError:
+                        event = from_json(msg, data_unmarshaller=pickle.loads)
+                    if self._get_ens_id(event["source"]) != self.ensemble.id_:
+                        logger.info(
+                            "Got event from evaluator "
+                            f"{self._get_ens_id(event['source'])} "
+                            f"with source {event['source']}, "
+                            f"ignoring since I am {self.ensemble.id_}"
+                        )
+                        continue
+                    try:
+                        await self._dispatcher.handle_event(event)
+                    except BaseException as ex:  # pylint: disable=broad-except
```

Comment:
This is not part of this PR, but handling `BaseException` does not look right.

Reply:
Agree. At least  `handle_event`  may raise `asyncio.InvalidStateError` but then also a pretty general ` await f(event) ` is involved here where  `f`  is a step (?).